DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Repositories/AcademyRepository.cs

```diff
@@ -82,20 +82,19 @@ await academiesDbContext.MisEstablishments
                         ConvertStringToSafeguardingScore(me.PreviousSafeguardingIsEffective),
                         me.PreviousInspectionStartDate.ParseAsNullableDate())))
                 .ToListAsync();
-
-        // Look in MisFurtherEducationEstablishments for academies not found in MisEstablishments
+        // Check to see if all ratings have been found in MisEstablishments, if not search in MisFurtherEducationEstablishments
         // Note: if an entry is in MisEstablishments then it will not be in MisFurtherEducationEstablishments, even if it has no ofsted data
-        var urnsNotInMisEstablishments = urns.Except(ofstedRatings.Select(a => a.Urn)).ToArray();
-        if (urnsNotInMisEstablishments.Length > 0)
+        if (urns.Length != ofstedRatings.Count)
         {
+            var urnsNotInMisEstablishments = urns.Except(ofstedRatings.Select(a => a.Urn)).ToArray();
             ofstedRatings.AddRange(
-                await academiesDbContext.MisFurtherEducationEstablishments
+                await academiesDbContext.FurtherEducationEstablishmentsFiat
                     .Where(mfe => urnsNotInMisEstablishments.Contains(mfe.ProviderUrn))
                     .Select(mfe =>
                         new AcademyOfstedRatings(
                             mfe.ProviderUrn,
                             new OfstedRating(
-                                ConvertNullableIntToOfstedRatingScore(mfe.OverallEffectiveness),
+                                ConvertNullableIntToOfstedRatingScore(mfe.OverallEffectiveness.ParseAsNullableInt()),
```

Comment:
I'm assuming that this db column has become a varchar so it can more closely mirror the behaviour of the non-further ed establishments. If so then we should use `ConvertOverallEffectivenessToOfstedRatingScore` like we do for establishments. Same comment for the `mfe.PreviousOverallEffectiveness` below too